### ISSUE TITLE
Python 3 crash on storing binary strings

### ISSUE THREAD

**Issue by timgraham** [linsomniac/python-memcached#80]
This test (adapted from Django's test suite) passes on Python 2, but fails on Python 3:

``` diff
diff --git a/tests/test_memcache.py b/tests/test_memcache.py
index 321dc7b..0415de9 100644
--- a/tests/test_memcache.py
+++ b/tests/test_memcache.py
@@ -137,6 +137,30 @@ class TestMemcache(unittest.TestCase):
         value = self.mc.get(key)
         self.assertEqual(value, 5)

+    def test_binary_string(self):
+        # Binary strings should be cacheable
+        from zlib import compress, decompress
+        value = 'value_to_be_compressed'
+        compressed_value = compress(value.encode())
+
+        # Test set
+        self.mc.set('binary1', compressed_value)
+        compressed_result = self.mc.get('binary1')
+        self.assertEqual(compressed_value, compressed_result)
+        self.assertEqual(value, decompress(compressed_result).decode())
+
+        # Test add
+        self.mc.add('binary1-add', compressed_value)
+        compressed_result = self.mc.get('binary1-add')
+        self.assertEqual(compressed_value, compressed_result)
+        self.assertEqual(value, decompress(compressed_result).decode())
+
+        # Test set_many
+        self.mc.set_multi({'binary1-set_many': compressed_value})
+        compressed_result = self.mc.get('binary1-set_many')
+        self.assertEqual(compressed_value, compressed_result)
+        self.assertEqual(value, decompress(compressed_result).decode())
+
     def test_ignore_too_large_value(self):
         # NOTE: "MemCached: while expecting[...]" is normal...
         key = 'keyhere'
```

```
======================================================================
ERROR: test_binary_string (__main__.TestMemcache)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_memcache3.py", line 53, in test_binary_string
    compressed_result = self.mc.get('binary1')
  File "/home/tim/code/python-memcached/memcache.py", line 1093, in get
    return self._get('get', key)
  File "/home/tim/code/python-memcached/memcache.py", line 1077, in _get
    return _unsafe_get()
  File "/home/tim/code/python-memcached/memcache.py", line 1065, in _unsafe_get
    value = self._recv_value(server, flags, rlen)
  File "/home/tim/code/python-memcached/memcache.py", line 1235, in _recv_value
    val = buf.decode('utf8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x9c in position 1: invalid start byte
```

**Comment by rubendura** [comment]
I just found the same issue. I am trying to store some gzip-ed content and when getting that key it fails trying to decode it as utf-8 (when I expected those bytes to be returned). It looks like the library assumes that any bynary content is an utf-8 encoded string.

**Comment by Sovetnikov** [comment]
Now is 2016, python 3 bytes storing still fails with "UnicodeDecodeError: 'utf-8' codec can't decode byte".
This pull seem to fix the problem with adding _FLAGS_TEXT, but is still not accepted.
https://github.com/linsomniac/python-memcached/pull/74

**Comment by jan-schulz-k24** [comment]
Easiest workaround: store the bytes in an tuple: `(b'bytes')`.

**Comment by timgraham** [comment]
Fixed in e1f72daabc011b83a8debb86e9e2b7ff6307a41c.
